modules/http4s/src/smithy4s/http4s/SimpleProtocolBuilder.scala

```diff
@@ -178,12 +199,35 @@ abstract class SimpleProtocolBuilder[P](
     def flatMapErrors(
         fe: PartialFunction[Throwable, F[Throwable]]
     ): RouterBuilder[Alg, F] =
-      new RouterBuilder(service, impl, fe, middleware)
+      new RouterBuilder(
+        service,
+        impl,
+        fe,
+        middleware,
+        encodeErrorsBeforeMiddleware
+      )
 
     def middleware(
         mid: ServerEndpointMiddleware[F]
     ): RouterBuilder[Alg, F] =
-      new RouterBuilder[Alg, F](service, impl, errorTransformation, mid)
+      new RouterBuilder[Alg, F](
+        service,
+        impl,
+        errorTransformation,
+        mid,
+        encodeErrorsBeforeMiddleware = false
+      )
+
+    def middlewareWithEncodedErrors(
```

Comment:
Sorry it was I meant wasn't clear : what I asked for you to add : 

``` 
def encodeErrorsBeforeMiddleware(value: Boolean) : RouterBuilder[Alg, F]
```

the `middleware` method should not change the value of the boolean

Reply:
@Baccata , no, it’s me who got it wrong
Thank you! will do as you proposed

Reply:
[done](https://github.com/disneystreaming/smithy4s/pull/1669/commits/9491398f83ecf141a931b52aabb25ca90efebaa2)